src/OAuth2Adapter.php

```diff
@@ -47,9 +54,19 @@ public function authenticate(ServerRequestInterface $request) : ?UserInterface
     {
         try {
             $result = $this->resourceServer->validateAuthenticatedRequest($request);
-            $userId = $result->getAttribute('oauth_user_id', false);
-            if (false !== $userId) {
-                return $this->generateUser($userId, []);
+            $userId = $result->getAttribute('oauth_user_id', null);
+            $clientId = $result->getAttribute('oauth_client_id', null);
+            if (isset($userId) || isset($clientId)) {
+                return ($this->userFactory)(
+                    $userId ?? $clientId ?? '',
```

Comment:
Since one or the other of these has to be set (see previous conditional), this can be just `$userId ?? $clientId`.